lib/ansible/modules/cloud/digital_ocean/digital_ocean_domain.py

```diff
@@ -79,151 +79,119 @@
+        return resp['domain_record']
 
-        if not domain:
-            domain = Domain.find(name=getkeyordie("name"))
 
+def core(module):
+    do_manager = DoManager(module)
+    state = module.params.get('state')
+
+    domain = do_manager.find()
+    if state == 'present':
         if not domain:
-            domain = Domain.add(getkeyordie("name"),
-                                getkeyordie("ip"))
-            module.exit_json(changed=True, domain=domain.to_json())
+            domain = do_manager.add()
+            if 'message' in domain:
+                module.fail_json(changed=False, msg=domain['message'])
+            else:
+                module.exit_json(changed=True, domain=domain)
         else:
-            records = domain.records()
+            records = do_manager.all_domain_records()
             at_record = None
-            for record in records:
-                if record.name == "@" and record.type == 'A':
+            for record in records['domain_records']:
+                if record['name'] == "@" and record['type'] == 'A':
                     at_record = record
 
-            if not at_record.data == getkeyordie("ip"):
-                record.update(data=getkeyordie("ip"), record_type='A')
-                module.exit_json(changed=True, domain=Domain.find(id=record.id).to_json())
-
-        module.exit_json(changed=False, domain=domain.to_json())
-
-    elif state in ('absent'):
-        domain = None
-        if "id" in module.params:
-            domain = Domain.find(id=module.params["id"])
-
-        if not domain and "name" in module.params:
-            domain = Domain.find(name=module.params["name"])
+            if not at_record['data'] == module.params.get('ip'):
+                do_manager.edit_domain_record()
+                module.exit_json(changed=True, domain=do_manager.find())
+            else:
+                module.exit_json(changed=False, domain=do_manager.domain_record())
 
+    elif state == 'absent':
         if not domain:
-            module.exit_json(changed=False, msg="Domain not found.")
-
-        event_json = domain.destroy()
-        module.exit_json(changed=True, event=event_json)
+            module.fail_json(changed=False, msg="Domain not found")
+        else:
+            delete_event = do_manager.destroy_domain()
+            if not delete_event:
+                module.fail_json(changed=False, msg=delete_event['message'])
+            else:
+                module.exit_json(changed=True, event=None)
+        delete_event = do_manager.destroy_domain()
+        module.exit_json(changed=delete_event)
 
 
 def main():
     module = AnsibleModule(
         argument_spec=dict(
             state=dict(choices=['present', 'absent'], default='present'),
-            api_token=dict(aliases=['API_TOKEN'], no_log=True),
+            oauth_token=dict(
+                aliases=['API_TOKEN'],
```

Comment:
This broke backwards compatibility for me.  I think it should be lowercase `'api_token'`?